File: Atcoder_#363/ruby/b_japanese_cursed_doll.rb
n, t, p = gets.split(" ").map(&:to_i)
l = gets.split(" ").map(&:to_i)

# 現在の髪の長さがT以上の人の数を数える
initial_count = l.count { |length| length >= t }

# すでに条件を満たしている場合は0を出力
if initial_count >= p
  puts 0
else
  # 条件を満たすために必要な日数を計算
  days = 0
  loop do
    days += 1
    new_count = l.count { |length| length + days >= t }
    if new_count >= p
      puts days
      break
    end
  end
end

File: Atcoder_#363/javascript/b_japanese_cursed_doll.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

let inputLines = [];
rl.on('line', (line) => {
  inputLines.push(line);
}).on('close', () => {
  processInput(inputLines);
});

function processInput(input) {
  const [n, t, p] = input[0].split(" ").map(Number);
  const l = input[1].split(" ").map(Number);

  // 現在の髪の長さがT以上の人の数を数える
  const initialCount = l.filter(length => length >= t).length;

  // すでに条件を満たしている場合は0を出力
  if (initialCount >= p) {
    console.log(0);
  } else {
    // 条件を満たすために必要な日数を計算
    let days = 0;
    while (true) {
      days += 1;
      const newCount = l.filter(length => length + days >= t).length;
      if (newCount >= p) {
        console.log(days);
        break;
      }
    }
  }
}
